volume controller: add configurable integration test to stress the binder

jsafrane opened the pull request:
The test tries to bind configured nr. of PVs to the same nr. of PVCs. '100' is used by default, which should take ~1-3 seconds (depends on log level). Periodic sync is needed in rare cases, which may add another 10 seconds. - cache from #25881 will help here and sync should not be needed at all.

The test is configurable and may be reused to measure binder performance. Set KUBE_INTEGRATION_PERSISTENTVOLUME_\* env. variables as described in persistent_volume_test.go and run the tests:

```
# compile
$ cd test/integration
$ godep go test -tags 'integration no-docker' -c

# run the tests
$ KUBE_INTEGRATION_PERSISTENTVOLUME_SYNC_PERIOD=10s KUBE_INTEGRATION_PERSISTENTVOLUME_OBJECTS=1000 time ./integration.test -test.run TestPersistentVolumeMultiPVsPVCs -v 2
```

Log level '2' is useful to get timestamps of various events like 'TestPersistentVolumeMultiPVsPVCs: start' and 'TestPersistentVolumeMultiPVsPVCs: claims are bound'.

jsafrane commented:
@k8s-bot test this issue:#26123

jsafrane commented:
cc: @kubernetes/sig-storage

jayunit100 commented:
@jsafrane   For scale can we use https://github.com/kubernetes/kubernetes/pull/24536 (`--scale`) option ?  the idea would be to define a base load and allow users to supply a generic scale option which multiplies the test load by an integral amount.  Thus it could apply the same way to the iperf, density, logging, and other scaling tests.

jsafrane commented:
@jayunit100, that sounds great! I'll reuse it for amount of created objects and use env. variables only for very specific tuning like controller sync interval.

jsafrane commented:
@k8s-bot test this issue: #26271

jsafrane commented:
`--scale` is not merged yet, I will leave the test as it is with env. variables. I'll rework it when `--scale` gets merged.

pmorie commented:
suggestion: s/PERSISTENTVOLUME/PV/g

jsafrane commented:
env. variables renamed to *_PV_*

jsafrane committed:
volume controller: add configurable integration test to stress the binder

The test tries to bind configured nr. of PVs to the same nr. of PVCs.
'100' is used by default, which should take ~1-3 seconds (depends on log level).
Periodic sync is needed in rare cases, which may add another 10 seconds. - cache
from #25881 will help here and sync should not be needed at all.

The test is configurable and may be reused to measure binder performance.
Set KUBE_INTEGRATION_PV_* env. variables as described in
persistent_volume_test.go and run the tests:

# compile
$ cd test/integration
$ godep go test -tags 'integration no-docker' -c

# run the tests
$ KUBE_INTEGRATION_PV_SYNC_PERIOD=10s KUBE_INTEGRATION_PV_OBJECTS=1000 time ./integration.test -test.run TestPersistentVolumeMultiPVsPVCs -v 2

Log level '2' is useful to get timestamps of various events like
'TestPersistentVolumeMultiPVsPVCs: start' and 'TestPersistentVolumeMultiPVsPVCs:
claims are bound'.

jsafrane commented:
Rebased

pmorie commented on test/integration/persistent_volumes_test.go:
```diff
@@ -239,12 +298,94 @@ func TestPersistentVolumeMultiPVs(t *testing.T) {
+	// Create the volumes in a separate goroutine to pop events from
+	// watchPV early - it seems it has limited capacity and it gets stuck
+	// with >3000 volumes.
+	go func() {
```
good idea, it was terribly slow to run this before
